File: Algorithms/C#/2404-most_frequent_even_element.cs
public class Solution {
    public int MostFrequentEven(int[] nums) {

        Dictionary<int, int> map = new Dictionary<int,int>();
        List<int> list = new List<int>();
        
        foreach(int i in nums){
            if(map.ContainsKey(i)){
                map[i]++;
            
            }else if(i % 2 == 0){
                map.Add(i, 1);
            }
        }

        if(map.Count == 0){
            return -1;
        }

        int a = int.MinValue;
        foreach(int i in map.Keys){
           if(map[i] > a){
            a = map[i];
            list.Clear();
            list.Add(i);
           
           }else if(map[list.ElementAt(0)] == map[i]){
            list.Add(i);
           }
        }

        if(list.Count == 0){
            return list.ElementAt(0);
        }else{
            return list.Min();
        }
        
    }
}

File: Algorithms/Python/2404-most_frequent_even_element.py
class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:

        myMap = {}
        key = -1
        for n in nums:
            if n % 2 != 0:
                continue 
            if n not in myMap:
                myMap[n] = 1
            else:
                myMap[n] += 1
            
            if key == -1 or myMap[n] > myMap[key] or (myMap[n] == myMap[key] and n < key):
                key = n
        
        if key == -1:
            return -1
        return key
            